### PR TITLE
Web/Amazon Search and Cart Functionality

### PR DESCRIPTION
**Pull request by snehith07**
Implemented Amazon Search and Cart Functionality

### CODE REVIEW

File: src/test/resources/com/znsio/e2e/features/amazon.feature [hunk 1]
```diff
@@ -0,0 +1,10 @@
+@amazon
+Feature: Amazon Search and Cart functionality test
+
+ @web
+  Scenario: Verify add to cart functionality with a Guest user
+    Given Guest user is on amazon home screen
+    When user searches for "iphone 13"
+    And opens the product details page
+    And Adds the product to cart
+    Then User should see the product in the cart
```
Comment (ShamaUgale) [inline src/test/resources/com/znsio/e2e/features/amazon.feature]: The steps can be combined. As an example:

Scenario: user can add product to cart
Given user searches for "iPhone 13" product on amazon landing page
When user selects first product from search results
And user adds product into cart
Then added product should be visible into cart

File: src/test/java/com/znsio/e2e/steps/AmazonSteps.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    @And("opens the product details page")
+    public void opensTheProductDetailsPage() {
+        new AmazonBL().openTheProductDetailsPage();
+    }
+
+    @And("Adds the product to cart")
+    public void addsTheProductToCart() {
+        new ProductDetailsBL().addTheProductToCart();
```
Comment (ShamaUgale) [inline src/test/java/com/znsio/e2e/steps/AmazonSteps.java]: userAddsTheProductToTheCart();

File: src/test/java/com/znsio/e2e/steps/AmazonSteps.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Given("Guest user is on amazon home screen")
+    public void guestUserIsOnAmazonHomeScreen() {
+        LOGGER.info("launch website as a guest user");
+        allDrivers.createDriverFor(GUEST_USER_CONTEXT.GUEST_USER, Runner.platform, context);
+    }
+
+    @When("user searches for {string}")
+    public void userSearchesFor(String product) {
+        new AmazonBL().searchForProduct(product);
```
Comment (ShamaUgale) [inline src/test/java/com/znsio/e2e/steps/AmazonSteps.java]: userSearchsForProduct(product)

File: src/test/java/com/znsio/e2e/steps/AmazonSteps.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import com.znsio.e2e.tools.Drivers;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.log4j.Logger;
+
+public class AmazonSteps {
+
+    private static final Logger LOGGER = Logger.getLogger(CalculatorSteps.class.getName());
```
Comment (ShamaUgale) [inline src/test/java/com/znsio/e2e/steps/AmazonSteps.java]: CalculatorSteps.class.getName()?

File: src/test/java/com/znsio/e2e/screen/web/ProductScreenWeb.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    driver.waitForClickabilityOf(addToCartButton);
+    driver.findElement(addToCartButton).click();
+    driver.waitTillElementIsPresent(addCartMessage);
+    return driver.findElement(addCartMessage).getText();
+    }
+
+    public ProductScreen goToCart(){
+        LOGGER.info("clicking on go to cart button");
+        driver.waitForClickabilityOf(goToCartButton);
+        driver.findElement(goToCartButton).click();
```
Comment (ShamaUgale) [inline src/test/java/com/znsio/e2e/screen/web/ProductScreenWeb.java:37]: line 35 and 36 can be combined

**Review by ShamaUgale — CHANGES_REQUESTED** [review]
Reviewed it half way. Please complete and change the implementations based on the scenario in feature file

File: src/test/java/com/znsio/e2e/screen/web/ProductScreenWeb.java (hunk 5 above)
Comment (snehith07) [reply src/test/java/com/znsio/e2e/screen/web/ProductScreenWeb.java:37]: When these steps are combined, 
`driver.waitForClickabilityOf(goToCartButton).findElement(goToCartButton).click();`
the code is failing due to an error: _"Unable to locate element"_

working fine if these steps are not combined.

**Pull request closed without merging**